ripple/src/main/test/script.test.js

```diff
@@ -24,4 +28,122 @@ test('convertToRawString: test #5 Converts to upper case w/o spaces or special c
 
 test('convertToRawString: test #6 Converts accented characters to ascii', () => {
   expect(scriptsModule.convertToRawString("á, é, í, ó, ú, ü, ñ, ¿, ¡")).toBe("AEIOUUN");
+});
+
+test('serve: test #6 Converts accented characters to ascii', () => {
+  expect(scriptsModule.convertToRawString("á, é, í, ó, ú, ü, ñ, ¿, ¡")).toBe("AEIOUUN");
+});
+
+/* Tests that the image element's source is properly set by function. */
+test('serveBlob: changes image source', () => {
+  // Set up our document body
+  document.body.innerHTML =
+    '<div>' +
+    '  <img id="test-img" src="old" />' +
+    '</div>';
+  // Expect that the url returned is accurate
+  scriptsModule.serveBlob("blobKey", "test-img");
+  expect(document.getElementById('test-img').src).toEqual("http://localhost/serve?blob-key=blobKey");
+});
+
+/* Tests that the function reads the url and retrieves the blob key correctly. */
+test('getParameterByName: reads the blob key from the url', () => {
+  // Set up window.href
+  global.window = Object.create(window);
+  const url = "http://localhost/serve?blob-key=blobKey";
+  Object.defineProperty(window, 'location', {
+      value: {
+      href: url
+    }
+  });
+  // Expect that the regex retrieves the correct parameter string
+  expect(scriptsModule.getParameterByName("blob-key")).toEqual("blobKey");
+}); 
+
+/* Tests that DOM is actually altered by function. */
+test('addTextToDom: modifies document inner html', () => {
+  // Set up our document body
+  document.body.innerHTML =
+    '<div>' +
```

Comment:
since this string is used a lot, you can probably just save it in a variable and just assign that variable to document.body.innerHtml in each test

Reply:
fixed